database-commons/src/main/java/io/cdap/plugin/db/ConnectionConfig.java

```diff
@@ -111,17 +132,55 @@ public Properties getConnectionArguments() {
     return arguments;
   }
 
+  /**
+   * Returns all configuration properties including database-specific ones as single string.
+   * In the case when there are no connection arguments, an empty string will be returned.
+   * @return connection arguments as string with '=' as key-value delimiter and ';' as pair delimiter.
```

Comment:
are these delimiters safe to use for every db type? You could also use Gson to serialize/deserialize and then you wouldn't have to worry about special characters.